backend/inspirehep/pidstore/providers/texkey.py

```diff
@@ -75,41 +86,20 @@ def get_texkey_with_random_part(cls, texkey):
                 return pid_value
         raise CannotGenerateUniqueTexKey
 
-    @classmethod
-    def query_pid_for_texkey(cls, texkey):
-        texkey = (
-            PersistentIdentifier.query.filter(PersistentIdentifier.pid_value == texkey)
-            .filter(PersistentIdentifier.pid_type == cls.pid_type)
-            .one_or_none()
-        )
-        return texkey
-
     @classmethod
     def query_all_texkeys_starting_with(cls, texkey):
         return {
             result[0]
-            for result in PersistentIdentifier.query.with_for_update()
-            .with_entities(PersistentIdentifier.pid_value)
+            for result in PersistentIdentifier.query.with_entities(
+                PersistentIdentifier.pid_value
+            )
             .filter(PersistentIdentifier.pid_type == cls.pid_type)
             .filter(PersistentIdentifier.pid_value.startswith(texkey))
         }
 
-    @classmethod
-    def is_texkey_assigned_correctly(cls, texkey, uuid):
-        texkey = cls.query_pid_for_texkey(texkey)
-        if (
-            texkey
-            and texkey.object_uuid == uuid
-            and texkey.status == PIDStatus.REGISTERED
-        ):
-            return True
-        elif texkey:
-            raise PIDAlreadyExistsError(texkey.pid_type, texkey.pid_value)
-        return False
-
     @classmethod
     def create_or_update_texkey_in_pidstore(cls, texkey, uuid):
```

Comment:
I didn't catch this earlier, but this is a misnomer: it doesn't actually create a texkey in the pidstore.

Reply:
:+1: